testing/trino-product-tests/src/main/java/io/trino/tests/product/utils/DeltaQueryExecutor.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.tests.product.utils;
+
+import com.google.common.collect.ImmutableList;
+import io.airlift.resolver.ArtifactResolver;
+import io.airlift.resolver.DefaultArtifact;
+import io.trino.tempto.context.TestContext;
+import io.trino.tempto.query.JdbcConnectionsPool;
+import io.trino.tempto.query.JdbcConnectivityParamsState;
+import io.trino.tempto.query.JdbcQueryExecutor;
+import org.sonatype.aether.artifact.Artifact;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.sql.Connection;
+import java.sql.Driver;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.List;
+import java.util.Properties;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static io.airlift.resolver.ArtifactResolver.MAVEN_CENTRAL_URI;
+import static io.airlift.resolver.ArtifactResolver.USER_LOCAL_REPO;
+import static java.lang.ClassLoader.getPlatformClassLoader;
+
+public class DeltaQueryExecutor
+        extends JdbcQueryExecutor
+{
+    private static final DeltaConnectionsPool CONNECTIONS_POOL = new DeltaConnectionsPool();
+
+    public DeltaQueryExecutor(TestContext testContext)
+    {
+        super(testContext.getDependency(JdbcConnectivityParamsState.class, "delta"), CONNECTIONS_POOL, testContext);
+    }
+
+    private static class DeltaConnectionsPool
+            extends JdbcConnectionsPool
+    {
+        private Connection connection;
+
+        @Override
+        public Connection connectionFor(JdbcConnectivityParamsState jdbcParamsState)
+                throws SQLException
+        {
+            if (connection == null || connection.isClosed()) {
+                Properties properties = new Properties();
+                properties.put("user", jdbcParamsState.user);
+                properties.put("password", jdbcParamsState.password);
+                connection = getDatabaseDriver(jdbcParamsState).connect(jdbcParamsState.url, properties);
```

Comment:
we should load the driver once and keep it on a field

```
private final Supplier<Driver> driver = Suppliers.memoize(::getDatabaseDriver);
```

Reply:
Changed to `Cache` since `getDatabaseDriver` expects one argument.